### PR TITLE
feat: legal hold UI - create bars and banners and handle clicks [WPB-5446]

### PR DESCRIPTION
**Pull request by saleniuk**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-5446" title="WPB-5446" target="_blank">Sub-taskWPB-5446</a>  [Android] UI - create all dialogs related to legal hold
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

In order to show the legal hold state for the currently logged in user and for any other profile or conversation details, we need to have proper bars and banners implemented.

### Solutions

Implemented `LegalHoldBar` which shows the legal hold status for the current user and add it to the `CommonTopAppBar`.
Implemented `LegalHoldSubjectBanner` which can be added to the other user profile screen or conversation details screen and shows that the user or one of the members is a subject of the legal hold.
Fixed dark theme top bar colors.

### Attachments (Optional)

image

https://github.com/wireapp/wire-android-reloaded/assets/30429749/d4550019-e5e6-465c-b36a-d23c88131be3

image

https://github.com/wireapp/wire-android-reloaded/assets/30429749/1f76677d-f9c8-4a07-afde-8f1bed2ad7a0

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1860](https://10.10.124.134/job/AR-build-pipeline/1860/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.6.0(8118058)(8118058)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.6.0(8118058)(8118058)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/ui/legalhold/banner/LegalHoldSubjectBanner.kt [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import com.wire.android.ui.common.LegalHoldIndicator
+import com.wire.android.ui.common.colorsScheme
+import com.wire.android.ui.common.dimensions
+import com.wire.android.ui.common.typography
+import com.wire.android.ui.theme.WireTheme
+import com.wire.android.util.ui.PreviewMultipleThemes
+import com.wire.android.util.ui.stringWithStyledArgs
+
+@Composable
+fun LegalHoldSubjectBanner(
```
Comment (borichellow) [inline app/src/main/kotlin/com/wire/android/ui/legalhold/banner/LegalHoldSubjectBanner.kt:46]: Is it needed? I don't see any usage of it

**Review by borichellow — APPROVED** [review]
nice job! 👍 
Jus a small question :)

**Review by ohassine — APPROVED** [review]

File: app/src/main/kotlin/com/wire/android/ui/legalhold/banner/LegalHoldSubjectBanner.kt (hunk 1 above)
Comment (saleniuk) [reply app/src/main/kotlin/com/wire/android/ui/legalhold/banner/LegalHoldSubjectBanner.kt:46]: This will be used in the next legal hold PRs 😃

**Pull request merged**